src/sinks/new_relic_logs.rs

```diff
@@ -0,0 +1,97 @@
+use crate::{
+    buffers::Acker,
+    sinks::http::{Encoding, HttpMethod, HttpSinkConfig},
+    sinks::util::{BatchConfig, Compression, TowerRequestConfig},
+    topology::config::{DataType, SinkConfig, SinkDescription},
+};
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone, Derivative)]
+#[serde(rename_all = "snake_case")]
+#[derivative(Default)]
+pub enum NewRelicLogsRegion {
+    #[derivative(Default)]
+    Us,
+    Eu,
+}
+
+#[derive(Deserialize, Serialize, Debug, Default, Clone)]
+pub struct NewRelicLogsConfig {
+    pub license_key: Option<String>,
+    pub insert_key: Option<String>,
+    pub region: Option<NewRelicLogsRegion>,
+
+    #[serde(default, flatten)]
+    pub batch: BatchConfig,
+
+    #[serde(flatten)]
+    pub request: TowerRequestConfig,
+}
+
+inventory::submit! {
+    SinkDescription::new::<NewRelicLogsConfig>("new_relic_logs")
+}
+
+#[typetag::serde(name = "new_relic_logs")]
+impl SinkConfig for NewRelicLogsConfig {
+    fn build(&self, acker: Acker) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let mut headers: IndexMap<String, String> = IndexMap::new();
+
+        if let Some(license_key) = &self.license_key {
+            headers.insert("X-License-Key".to_owned(), license_key.clone());
+        } else if let Some(insert_key) = &self.insert_key {
+            headers.insert("X-Insert-Key".to_owned(), insert_key.clone());
+        } else {
+            return Err(format!("must provide either 'license_key' or 'insert_key'").into());
+        }
+
+        let uri = match self.region.as_ref().unwrap_or(&NewRelicLogsRegion::Us) {
+            NewRelicLogsRegion::Us => "https://log-api.newrelic.com/log/v1",
+            NewRelicLogsRegion::Eu => "https://log-api.eu.newrelic.com/log/v1",
+        };
+
+        let batch_conf = BatchConfig {
+            // The max request size is 10MiB, so in order to be comfortably
```

Comment:
Hmm, I feel like we should at least try to push more than -50% of the batch size? Ideally, we should try at least 8mb, maybe more? I can't remember if our batching will add more or not but this would be a good thing to verify. 5mb seems a bit low.

Reply:
I'd rather keep it conservative for now. There's hopefully going to be some internal stress tests done soon and then we can take another look at maybe increasing the defaults.

Reply:
I'm indifferent but tend to agree that we should take full advantage of the payload size if possible. This should only kick in if volume is high enough, especially with a 1 second timeout. New Relic is bound to have some very large customers that will benefit from this.